lib/cisco_node_utils/bgp_neighbor_af.rb

```diff
@@ -544,23 +557,49 @@ def default_route_reflector_client
     end
 
     # -----------------------
-    # <state> send-community [ both | extended | standard ]
-    # NOTE: 'standard' is default but does not nvgen on some platforms
-    # Returns: none, both, extended, or standard
+    # Nexus and XR implementations of send community differ enough that it makes
+    # sense to split up the individual methods to support them
     def send_community
       val = config_get('bgp_neighbor_af', 'send_community', @get_args)
       return default_send_community if val.nil?
+      platform == :nexus ? send_comm_nexus_get(val) : send_comm_ios_xr_get(val)
+    end
+
+    # Nexus: <state> send-community [ both | extended | standard ]
+    #  NOTE: 'standard' is default but does not nvgen on some platforms
+    #  Returns: none, both, extended, or standard
+    def send_comm_nexus_get(val)
       val = val.split.last
       return 'standard' if val[/send-community/] # Workaround
       val
     end
 
+    # XR: 'send-community-ebgp' and 'send-extended-community-ebgp' are
+    #  the equivalents of the NXOS: standard | extended functionality
+    #  Returns: node, 'send-community-ebgp', 'send-extended-community-ebgp' or
+    #  'send-community-ebgp send-extended-community-ebgp' which is the 'both'
+    # keyword equivalent
+    def send_comm_ios_xr_get(val)
+      if val == ['send-community-ebgp', 'send-extended-community-ebgp']
+        val = 'both'
+      else
+        # Properties are an array
+        val = val.join
```

Comment:
So I've learned something here I didn't know before (idempotency etc). If I do this, then because the getter is basically the same it means I can share a lot more code between the two platforms - especially the minitest code.

Without this change I was converting ['send-community-ebgp', 'send-extended-community-ebgp'] to 'both' in the setter - but no need now.

Tested - seems to work better.